src/async/ExtensiblePromise.ts

```diff
@@ -46,12 +46,33 @@ export default class ExtensiblePromise<T> {
 	}
 
 	/**
-	 * Return a ExtensiblePromise that resolves when all of the passed in objects have resolved
+	 * Return a ExtensiblePromise that resolves when all of the passed in objects have resolved. When used with a key/value
+	 * pair, the returned promise's argument is a key/value pair of the original keys with their resolved values.
 	 *
-	 * @param iterable    An iterable of values to resolve. These can be Promises, ExtensiblePromises, or other objects
+	 * @example
+	 * ExtensiblePromise.all({ one: 1, two: 2 }).then(results => console.log(results));
+	 * // { one: 1, two: 2 }
+	 *
+	 * @param iterable    An iterable of values to resolve, or a key/value pair of values to resolve. These can be Promises, ExtensiblePromises, or other objects
 	 * @returns {ExtensiblePromise}
 	 */
-	static all<F extends ExtensiblePromise<T>, T>(iterable: Iterable<(T | Thenable<T>)> | (T | Thenable<T>)[]): F {
+	static all<F extends ExtensiblePromise<T>, T>(iterable: { [_: string]: T | Promise<T> | Thenable<T> } | Iterable<(T | Thenable<T>)> | (T | Thenable<T>)[]): F {
```

Comment:
They are different types? The parameter is taking `{ [_: string]: T | Promise<T> | Thenable<T> }` and the return value is just `{ [_: string]: T }`.

One thing we can do (although it has nothing to do with this change) is break out the definition,

```ts
static all<F extends ExtensiblePromise<{ [_: string]: T }>, T>(iterable: { [_: string]: T | Promise<T> | Thenable<T> }): F;
static all<F extends ExtensiblePromise<T[]>, T>(iterable: Iterable<(T | Thenable<T>)> | (T | Thenable<T>)[]): F;
static all<F extends ExtensiblePromise<T>, T>(iterable: { [_: string]: T | Promise<T> | Thenable<T> } | Iterable<(T | Thenable<T>)> | (T | Thenable<T>)[]): F {
```

That would at least give a little bit more definition to the return type?